sdk/spring/azure-spring-boot/src/main/java/com/microsoft/azure/spring/autoconfigure/aad/AzureADGraphClient.java

```diff
@@ -121,22 +121,21 @@ private static String getResponseStringFromConn(HttpURLConnection conn) throws I
     }
 
     private List<UserGroup> loadUserGroups(String graphApiToken) throws IOException {
-        final String responseInJson = getUserMembershipsV1(graphApiToken);
+        String responseInJson = getUserMemberships(graphApiToken, Optional.empty());
         final List<UserGroup> lUserGroups = new ArrayList<>();
         final ObjectMapper objectMapper = JacksonObjectMapperFactory.getInstance();
-        final JsonNode rootNode = objectMapper.readValue(responseInJson, JsonNode.class);
-        final JsonNode valuesNode = rootNode.get("value");
-
-        if (valuesNode != null) {
-            lUserGroups
-                .addAll(StreamSupport.stream(valuesNode.spliterator(), false).filter(this::isMatchingUserGroupKey)
-                    .map(node -> {
-                        final String objectID = node.
-                            get(aadAuthenticationProperties.getUserGroup().getObjectIDKey()).asText();
-                        final String displayName = node.get("displayName").asText();
-                        return new UserGroup(objectID, displayName);
-                    }).collect(Collectors.toList()));
+        objectMapper.registerModule(new Jdk8Module());
+        UserGroups groupsFromJson = objectMapper.readValue(responseInJson, UserGroups.class);
 
+        if (groupsFromJson.getValue() != null) {
+            lUserGroups.addAll(groupsFromJson.getValue().stream().filter(this::isMatchingUserGroupKey)
+                .collect(Collectors.toList()));
+        }
+        while (groupsFromJson.getOdataNextLink().isPresent()) {
```

Comment:
I debugged in my localhost, `groupsFromJson.getOdataNextLink()` will return `null`, not `Optional.empty()`, so `groupsFromJson.getOdataNextLink().isPresent()` will throw `NullPointerExceotion`.

Json:

![image](https://user-images.githubusercontent.com/13167207/91273539-fee5fd80-e7af-11ea-967a-edefd156b7ba.png)

Screenshot of IDE:
![2020-08-26 152522](https://user-images.githubusercontent.com/13167207/91273988-9fd4b880-e7b0-11ea-856b-0656795ac355.png)

Reply:
thanks for the debugging. The change from `Optional<String>`  to `String` means that `getOdataNextLink()` now returns a string or null if there is no `odata.nextLink` in the response. I replaced `isPresent()` with a null check in https://github.com/Azure/azure-sdk-for-java/pull/14305/commits/cc20c0521dbed11c5ffc661c23de8ed1ff65e9b4